### PR TITLE
Window dnn

### PR DESCRIPTION
**Pull request by georgeSoton**

### CODE REVIEW

File: data/pipelines.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        '\'',
+        '\'\'',
+        'would',
+        'could',
+        '\'s',
+        'one']
+
+    stops = [*stopwords.words('english'), *extra_stopwords]
+    for s in prior:
+        yield [
```
Comment (superdyoll) [inline data/pipelines.py:36]: Is it worth adding string.punctuation into this aswell as the stopwords list does not include punctuation? Could add this line in s = s.translate(None, string.punctuation) in here.

File: Predictors/probabilistic_with_fill.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        self.cull_stopwords = cull_stopwords
+
+    def fit(self, sentences, labels):
+        distinct_labels = set(labels)
+        for l in distinct_labels:
+            self.counterTable[l] = Counter()
+
+        for s, l in zip(sentences, labels):
+            # Strip punctuation
+            words = tokenize(
```
Comment (superdyoll) [inline Predictors/probabilistic_with_fill.py]: import_data.tokenize only takes in one argument, this one provides 3.

File: Predictors/probabilistic_with_fill.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    def predict(self, X):
+        try:
+            getattr(self, 'trained_')
+        except AttributeError:
+            raise RuntimeError('You must train the classifier before using it')
+
+        return [self.predict_sen_(s) for s in X]
+
+    def predict_sen_(self, s):
+        s = tokenize(
```
Comment (superdyoll) [inline Predictors/probabilistic_with_fill.py]: import_data.tokenize only takes in one argument, this one provides 3

File: Predictors/proxmarkov.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+        for l in distinct_labels:
+            self.trans[l] = dict()
+        count = 0
+        for s, l in zip(sentences, labels):
+            if count % 50 == 0:
+                print("Training: ", count)
+            count += 1
+
+            tokens = tokenize(s, stop=False)
```
Comment (superdyoll) [inline Predictors/proxmarkov.py]: import_data.tokenize only takes in 1 argument this one provides 2

File: Predictors/proxmarkov.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+            out = get_close_matches(w, vocab, n=n)[0]
+        except IndexError:
+            out = random.choice(vocab)
+        return out
+
+    def _sen_prob(self, s, l):
+        if self.runcount % 25 == 0:
+            print("RUNS:", self.runcount)
+        self.runcount += 1
+        tokens = tokenize(s, stop=False)
```
Comment (superdyoll) [inline Predictors/proxmarkov.py]: import_data.tokenize only takes in 1 argument but this one provides 2.

File: rnn_param_search.txt [hunk 6]
```diff
@@ -0,0 +1,47 @@
+Wednesday 02/05
```
Comment (superdyoll) [inline rnn_param_search.txt:1]: Does this file need to be in the repository?

**Review by superdyoll — CHANGES_REQUESTED** [review]
It seems you rewrote tokenize at some point but some of the classes try to give it more parameters than it can accept. Are you able to check that they still do what they were originally meant to do?

File: data/pipelines.py (hunk 1 above)
Comment (georgeSoton) [reply data/pipelines.py:36]: No, some punctuation is pretty unique.

File: rnn_param_search.txt (hunk 6 above)
Comment (superdyoll) [reply rnn_param_search.txt:1]: @tblah do you know if this file needs to be in the repo?

Comment (superdyoll) [reply rnn_param_search.txt:1]: I'm going to accept this merge request so we can get the code in and we can always remove this later.

**Review by superdyoll — APPROVED** [review]

**Pull request merged**